library/gui/screen/src/main/java/org/quiltmc/qsl/screen/api/client/ScreenMouseEvents.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright 2016, 2017, 2018, 2019 FabricMC
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.screen.api.client;
+
+import java.util.Objects;
+
+import org.quiltmc.qsl.base.api.event.Event;
+import org.quiltmc.qsl.base.api.event.client.ClientEventAwareListener;
+import org.quiltmc.qsl.screen.impl.client.ScreenExtensions;
+
+import net.minecraft.client.gui.screen.Screen;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+/**
+ * Events related to use of the mouse in a {@link Screen}.
+ *
+ * <p>All of these events work on top of a specific screen instance.
+ * Subscriptions will only last as long as the screen itself, they'll disappear once the screen gets refreshed, closed or replaced.
+ * Use {@link ScreenEvents#BEFORE_INIT} to register the desired events every time it is necessary.
+ *
+ * <p>Events are fired in the following order:
+ * <pre>{@code AllowX -> BeforeX -> AfterX}</pre>
+ * If the result of the Allow event is false, then Before and After are not called.
+ *
+ * @see ScreenEvents
+ */
+@Environment(EnvType.CLIENT)
+public final class ScreenMouseEvents {
```

Comment:
None of them can be `ClientEventAwareListener`!

Reply:
Oops, how didn't i see that lol

Reply:
Fixed.